### PR TITLE
Backport missing PRs from Bitcoin 0.13 - Part 4

### PR DESCRIPTION
**Pull request by codablock**
Continuation of https://github.com/dashpay/dash/pull/1770

### CODE REVIEW

File: src/qt/transactionview.cpp [hunk 1]
```diff
@@ -219,7 +219,7 @@ void TransactionView::setModel(WalletModel *model)
         transactionView->setSelectionBehavior(QAbstractItemView::SelectRows);
         transactionView->setSelectionMode(QAbstractItemView::ExtendedSelection);
         transactionView->setSortingEnabled(true);
-        transactionView->sortByColumn(TransactionTableModel::Status, Qt::DescendingOrder);
+        transactionView->sortByColumn(TransactionTableModel::Date, Qt::DescendingOrder);
```
Comment (UdjinM6) [inline src/qt/transactionview.cpp]: To match 8014 the same fix should be applied to the filter in `OverviewPage::SetupTransactionList`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -61,13 +75,13 @@ Refactoring is a necessary part of any software project's evolution. The followi
...
 
-Project maintainers aim for a quick turnaround on refactoring pull requests, so where possible keep them short, uncomplex and easy to verify. 
+Project maintainers aim for a quick turnaround on refactoring pull requests, so where possible keep them short, uncomplex and easy to verify.
 
 
 "Decision Making" Process
 -------------------------
 
-The following applies to code changes to the Dash Core project (and related projects such as libsecp256k1), and is not to be confused with overall Dash Network Protocol consensus changes.
+The following applies to code changes to the Dash Core project (and related projects such as libsecp256k1), and is not to be confused with overall Bitcoin Network Protocol consensus changes.
```
Comment (UdjinM6) [inline CONTRIBUTING.md]: this piece should be reverted

File: src/key.cpp [hunk 3]
```diff
@@ -124,9 +124,8 @@ bool CKey::Check(const unsigned char *vch) {
 }
 
 void CKey::MakeNewKey(bool fCompressedIn) {
-    RandAddSeedPerfmon();
     do {
-        GetRandBytes(vch, sizeof(vch));
+        GetStrongRandBytes(vch, sizeof(vch));
```
Comment (UdjinM6) [inline src/key.cpp]: Could probably be a good idea to use `GetStrongRandBytes` in `CMnemonic::Generate` too

File: doc/benchmarking.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+Benchmarking
+============
+
+Bitcoin Core has an internal benchmarking framework, with benchmarks
```
Comment (UdjinM6) [inline doc/benchmarking.md]: This file should be dashified

File: doc/benchmarking.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+for cryptographic algorithms such as SHA1, SHA256, SHA512 and RIPEMD160. As well as the rolling bloom filter.
```
Comment (UdjinM6) [inline doc/benchmarking.md]: Side note/TODO: maybe we should add a benchmark for X11 as well

**Comment by codablock** [comment]
Pushed fixes.

Build is still failing for various reasons. One was a timeout due to stdout/stderr buffers running full, added Bitcoin https://github.com/bitcoin/bitcoin/pull/8750 to fix this. Awaiting build result.

Another reason is that for some reason redirect_stderr is sometimes completely ignored, making the tests still fail...investigating.

I created 2 issues to not forget about the TODOs:
https://github.com/dashpay/dash/issues/1805
https://github.com/dashpay/dash/issues/1806

File: doc/benchmarking.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
+Benchmarking
+============
+
+Dash Core has an internal benchmarking framework, with benchmarks
+for cryptographic algorithms such as SHA1, SHA256, SHA512 and RIPEMD160. As well as the rolling bloom filter.
+
+After compiling dash-core, the benchmarks can be run with:
+`src/bench/bench_bitcoin`
```
Comment (UdjinM6) [inline doc/benchmarking.md]: Missed one, should be `src/bench/bench_dash`

Comment (codablock) [reply doc/benchmarking.md]: Fixed :+1:

**Comment by UdjinM6** [comment]
Slightly tested, ACK

**Pull request merged**